### ISSUE TITLE
Extend index definition in Defined Schema

### ISSUE THREAD

**Issue by mtrezza** [parse-community/parse-server#8764]
### New Feature / Enhancement Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Current Limitation

In Defined Schema, only simple compound indexes can be defined, the following index options cannot be defined:
- [partialFilterExpression](https://www.mongodb.com/docs/manual/core/index-partial/)
- [sparse](https://www.mongodb.com/docs/manual/core/index-sparse/)
- [ttl](https://www.mongodb.com/docs/manual/core/index-ttl/)
- [unique](https://www.mongodb.com/docs/manual/core/index-unique/)

### Feature / Enhancement Description

Add support for the above mentioned index options. These options do not have to be added explicitly, instead a general field ofr index options should be added to the index definition.

### Example Use Case

To avoid a breaking change, index options could be added by using an array instead of an object:

```
indexes: {
  index_without_option: {
    location: "2dsphere"
  },
  index_with_option: [{
    location: "2dsphere"
  }, {
    sparse: true
  }]
}
```

Alternatively for type consistency with breaking change:

```
indexes: {
  index_without_option: {
    fields: {
      location: "2dsphere"
    }
  },
  index_with_option: {
    fields: {
      location: "2dsphere"
    },
    options: {
      sparse: true
    }
  }
}
```

**Comment by mtrezza** [comment]
cc @Moumouls maybe something you'd be interested in

**Comment by Moumouls** [comment]
Yes @mtrezza, actually it was not implemented in defined schema because it's also not supported by the Parse.Schema. Defined schema rely on Parse.Schema, so i would say that's not really a defined schema issue, but a capability issue of the index endpoint (maybe rest or sdk)

**Comment by mtrezza** [comment]
Got it; Parse Server internally is capable of setting TTL and unique indexes, so I guess the Parse Server API just needs to be extended. I'll leave this issue open to add the feature to Defined Schema once it's added to Parse Schema.

**Comment by mtrezza** [comment]
Increased the bounty, given that this needs more work.

**Comment by badboy-tian** [comment]
@Moumouls hi, can you please toke look this question? thanks
https://community.parseplatform.org/t/why-so-slow-in-dashboard/3770/3
